.github/workflows/integration_tests.yml

```diff
@@ -771,56 +625,132 @@ jobs:
+            // Create properly formatted final test report
+            Object.entries(testResultMap).forEach(([test, testResults]) => {
+                const finalResult = {};
+                finalResult.test = test;
+                testResults.forEach(result => {
+                    const {test, ...results} = result;
+                    Object.assign(finalResult, results);
+                    
+                    const currentPlatform = Object.keys(results)[0];
+                    if(!allPlatforms.includes(currentPlatform)) {
+                        allPlatforms.push(currentPlatform);
+                    }
+                });
+                finalTestReport.push(finalResult);
+            });
+            
+            // Fill in platforms where a test was not executed
+            finalTestReport.forEach(testResult => {
+                allPlatforms.forEach(platform => {
+                    if(!(platform in testResult)) {
+                        testResult[platform] = "skip";
+                    }
+                });
+            });
+            
+            if(!fs.existsSync(FINAL_TEST_REPORTS_FOLDER)) {
+                fs.mkdirSync(FINAL_TEST_REPORTS_FOLDER);
+            }
+            fs.writeFileSync(FINAL_TEST_REPORT_FILEPATH_JSON, JSON.stringify(finalTestReport));
 
-      - name: Print Test Report
-        id: print_test_report
+      - name: Generate Markdown Test Report
         run: |
-          echo "Integration Tests have finished." > final-test-report.txt
-          echo "" >> final-test-report.txt
-          # please check with output in write_test_report task roughly 50 lines above
-          echo "| Target Platform        |  Keptn Install    |  Keptn Auth  | Airgapped Images | Log Ingestion   | Log Forwarding   | Sequence States   | Sequence Timeout   | Sequence Control   | Sequence Queue   | Sequence Loop |  QG Standalone  |  Self Healing  |  Del-Assist  |  Backup-Restore  |  Resource Service  |  User-Man. Depl.  | Cont Deliv | Cont Deliv (Remote Exec) | Manage secrets | Webhook integration | Graceful Shutdown | Uniform Registration   |" >> final-test-report.txt
-          echo "| ---------------------- | ----------------- | ------------ | ---------------- | --------------- | ---------------- | ----------------- | ------------------ | ---------------- | ------------------ | ------------- | --------------- | -------------- | ------------ | ---------------- | ------------------ | ----------------- |----------- | ------------------------ | -------------- | ------------------- | ----------------- | ---------------------- |" >> final-test-report.txt
-          cat test-report*.txt >> final-test-report.txt
-
-          # print test report
-          REPORT=$(cat final-test-report.txt)
-          echo $REPORT
-
-      - name: Upload final test report as an artifact
+          npx tablemark-cli@v2.0.0 "${{ env.FINAL_TEST_REPORT_FILEPATH_JSON }}" > "${{ env.FINAL_TEST_REPORT_FILEPATH_MARKDOWN }}" -N -c Test
```

Comment:
This step uses the `tablemark` CLI to generate a markdown table from the final JSON test report